src/pages/index.js

```diff
@@ -0,0 +1,327 @@
+    });
+  }
+
+  function getCardElement(data) {
+    const cardElement = cardTemplate.cloneNode(true).querySelector('.card');
+    const cardImage = cardElement.querySelector('.card__image');
+    const cardTitle = cardElement.querySelector('.card__title');
+    const cardLikeBtn = cardElement.querySelector('.card__like-btn');
+    const cardDeleteBtn = cardElement.querySelector('.card__delete-btn');
+
+    cardElement.dataset.cardId = data._id;
+
+
+    if (data.isLiked) {
+      cardLikeBtn.classList.add('card__like-btn_liked');
+    } else {
+      cardLikeBtn.classList.remove('card__like-btn_liked');
+    }
+
+    cardLikeBtn.addEventListener('click', () => {
+      if (cardLikeBtn.classList.contains('card__like-btn_liked')) {
+        api.unlikeCard(data._id)
+          .then(updatedCard => {
+            cardLikeBtn.classList.remove('card__like-btn_liked');
+          })
+          .catch(err => console.error('Error unliking card:', err));
+      } else {
+        api.likeCard(data._id)
+          .then(updatedCard => {
+            cardLikeBtn.classList.add('card__like-btn_liked');
+          })
+          .catch(err => console.error('Error liking card:', err));
+      }
+    });
+
+    cardImage.addEventListener('click', () => {
+      openModal(previewModal);
+      previewModalImage.src = data.link;
+      previewModalImage.alt = data.name;
+      previewModalCaption.textContent = data.name;
+    });
+
+    cardDeleteBtn.addEventListener('click', () => {
+      openDeleteModal(cardElement);
+    });
+
+    cardImage.src = data.link;
+    cardImage.alt = data.name;
+    cardTitle.textContent = data.name;
+
+    return cardElement;
+  }
+
+
+
+  function handleEditProfileSubmit(evt) {
+    evt.preventDefault();
+    const name = editModalNameInput.value;
+    const about = editModalDescriptionInput.value;
+
+    const saveButton = editProfileForm.querySelector('.modal__submit-btn');
+    saveButton.textContent = 'Saving...';
+    saveButton.disabled = true;
+
+    api.editUserProfile(name, about)
+      .then((userData) => {
+        profileName.textContent = userData.name;
+        profileDescription.textContent = userData.about;
+        closeModal(editProfileModal);
+      })
+      .catch((err) => {
+        console.error('Error updating profile:', err);
+      })
+      .finally(() => {
+        saveButton.textContent = 'Save';
+        saveButton.disabled = false;
+      });
+  }
+
+
+  function handleNewPostSubmit(evt) {
+    evt.preventDefault();
+    const name = newPostCaptionInput.value;
+    const link = newPostLinkInput.value;
+
+    api.createCard(name, link)
+      .then((newCardData) => {
+        const cardElement = getCardElement(newCardData);
+        cardsContainer.prepend(cardElement);
+        closeModal(newPostModal);
+        newPostForm.reset();
+        disableButton(cardSubmitBtn, settings);
+      })
+      .catch((err) => {
+        console.error('Error creating new card:', err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors